File: BidWinKoModify/app/src/main/java/com/bidwinko/screens/activity/SplashActivity.java
package com.bidwinko.screens.activity;

import android.app.Activity;
import android.app.AlertDialog;
import android.app.ProgressDialog;
import android.content.ActivityNotFoundException;
import android.content.Context;
import android.content.DialogInterface;
import android.content.Intent;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.net.Uri;
import android.os.Bundle;
import android.os.Handler;
import android.util.Log;
import android.widget.TextView;
import android.widget.Toast;

import com.bidwinko.BuildConfig;
import com.bidwinko.MainActivity;
import com.bidwinko.R;
import com.bidwinko.API.APIService;
import com.bidwinko.model.AppOpenModel;
import com.bidwinko.utilies.Constants;
import com.bidwinko.API.Retrofit;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class SplashActivity extends Activity {

    ProgressDialog progressDialog;
    String TAG = "Splash Screen" ,tranTxt ="";
    TextView txtVersion;


    @Override
    protected void onStart() {
        super.onStart();
        Log.e("SplashActivity","openig SplashActivity");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.e("SplashActivity","in OnDestroy  SplashActivity");
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        int versionCode = BuildConfig.VERSION_CODE;
        String versionName = BuildConfig.VERSION_NAME;

        txtVersion = findViewById(R.id.txtvesion);
        txtVersion.setText("Version: "+versionName);

        if(isNetworkAvailable(SplashActivity.this)){
            if (Constants.getSharedPreferenceInt(SplashActivity.this, "userId", 0)!=0 &&
                    !Constants.getSharedPreferenceString(SplashActivity.this, "securitytoken", "").equals("null"))
            {
                Log.e("SplashSCreen","openig OpenApp API");
//                openApp();

//                if (getIntent().getExtras() != null) {
//                    String title = null;
//                    String actionk = null;
//
//                    for (String key : getIntent().getExtras().keySet()) {
//                        if (key.equals("OfferKey")) {
//                            actionk = "True";
//                            title = getIntent().getExtras().getString(key);
//                        }
//                    }
//                    if(actionk!=null && actionk.equalsIgnoreCase("True")){
//                        Intent intentProductdDtails = new Intent(SplashActivity.this, ProductDetailsActivity.class);
//                        intentProductdDtails.putExtra("offerId", Integer.parseInt(title));
//                        startActivityForResult(intentProductdDtails,105);
//                    }
//                    else{
//                        openApp();
//                    }
//                }
//                else{
//                    openApp();
//                }

            } else {
                new Handler().postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        if(!((Activity) SplashActivity.this).isFinishing()){
                            progressDialog = new ProgressDialog(SplashActivity.this);
                            progressDialog.setMessage(getString(R.string.loadingwait));
                            progressDialog.show();
                            progressDialog.setCancelable(false);
                        }


                        Log.e("SplashSCreen","sending to google signin activity !");
                        Intent intent = new Intent(SplashActivity.this, GoogleLoginActivity.class);
                        startActivity(intent);
                        finish();

                    }
                },3000);

            }
        }
        else{
            android.app.AlertDialog.Builder builder = new AlertDialog.Builder(SplashActivity.this);
            builder.setTitle("Alert!");
            builder.setMessage("Please Check Your Internet Connection.");
            builder.setPositiveButton("Exit",
                    new DialogInterface.OnClickListener() {
                        public void onClick(DialogInterface dialog,
                                            int which) {
                            dialog.dismiss();
                            finish();
                        }
                    });
            builder.setCancelable(false);
            builder.show();
        }


    }

//    public void openApp(){
//        int versionCode = BuildConfig.VERSION_CODE;
//        final String versionName = BuildConfig.VERSION_NAME;
//
//        Constants.setSharedPreferenceInt(SplashActivity.this,"versionCode",versionCode);
//        Constants.setSharedPreferenceString(SplashActivity.this,"versionName",versionName);
//
//        APIService apiService = Retrofit.getClient().create(APIService.class);
//        Call<AppOpenModel> call = apiService.appOpen(Constants.getSharedPreferenceInt(SplashActivity.this,"userId",0),
//                Constants.getSharedPreferenceString(SplashActivity.this,"securitytoken",""),
//                Constants.getSharedPreferenceString(SplashActivity.this,"versionName",""),
//                Constants.getSharedPreferenceInt(SplashActivity.this,"versionCode",0),
//                Constants.getSharedPreferenceString(SplashActivity.this,"userFrom","") );
//
//        if(!((Activity) SplashActivity.this).isFinishing()) {
//            progressDialog = new ProgressDialog(SplashActivity.this);
//            progressDialog.setMessage(getString(R.string.loadingwait));
//            progressDialog.show();
//            progressDialog.setCancelable(false);
//        }
//
//
//        call.enqueue(new Callback<AppOpenModel>() {
//            @Override
//            public void onResponse(Call<AppOpenModel> call, Response<AppOpenModel> response) {
////                progressDialog.dismiss();
//                dismissProgressDialog();
////                Log.e("testing", "onResponse:112121spl "+response.toString() );
//                try {
//                    if(response!=null){
//                        if (response.isSuccessful()) {
////                    Log.e("testing", "onResponse:1forceupdate "+response.body().getForceUpdate() +" status: "+response.body().getStatus() );
//                            if (response.body().getForceUpdate()) {
//                                android.app.AlertDialog.Builder builder = new AlertDialog.Builder(SplashActivity.this);
//                                builder.setMessage("Your " + getString(R.string.app_name) + " seems very old, Please update to get new Earning features!!");
//                                builder.setPositiveButton("UPDATE NOW",
//                                        new DialogInterface.OnClickListener() {
//                                            public void onClick(DialogInterface dialog,
//                                                                int which) {
//                                                dialog.dismiss();
//
//                                                try {
//                                                    startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("market://details?id=" + getPackageName())));
//                                                } catch (ActivityNotFoundException e) {
//                                                    // TODO: handle exception
//                                                    startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("https://play.google.com/store/apps/details?id=" + getPackageName())));
//
//                                                }
//                                                finish();
//                                            }
//                                        });
//                                builder.setCancelable(false);
//                                builder.show();
//
//                            } else {
//                                if (response.body().getStatus() == 200) {
//
//                                    Log.e("SplashSCreen","response is 200");
//                                    String amount = String.valueOf(response.body().getAmount());
//                                    String coins = String.valueOf(response.body().getCoin());
//                                    String curency = String.valueOf(response.body().getCurrency());
//                                    int payoutlimit = response.body().getPaycoinLimit();
//                                    String FBurl = response.body().getFbLike();
//                                    String TWurl = response.body().getTwLike();
//                                    String TGMurl = response.body().getTelLike();
//
//                                    String bidremain = response.body().getBidBalance();
//                                    Constants.setSharedPreferenceString(SplashActivity.this,"bidBalance",bidremain);
//
//                                    Constants.setSharedPreferenceInt(SplashActivity.this, "paycoinLimit", payoutlimit);
//                                    Constants.setSharedPreferenceString(SplashActivity.this, "totalamount", amount);
//                                    Constants.setSharedPreferenceString(SplashActivity.this, "totalcoins", coins);
//                                    Constants.setSharedPreferenceString(SplashActivity.this, "curency", curency);
//
//                                    Constants.setSharedPreferenceString(SplashActivity.this, "fburl", FBurl);
//                                    Constants.setSharedPreferenceString(SplashActivity.this, "twurl", TWurl);
//                                    Constants.setSharedPreferenceString(SplashActivity.this, "tgmurl", TGMurl);
//
//                                    tranTxt = response.body().getTransText();
//                                    Constants.setSharedPreferenceString(SplashActivity.this, "tranTxt", tranTxt);
//
//                                    Intent intent = new Intent(SplashActivity.this, MainActivity.class);
//                                    startActivity(intent);
//                                    finish();
//
//                                } else {
//
//                                    Log.e("SplashSCreen","response is "+ response.body().getMessage());
//                                    Toast.makeText(SplashActivity.this, getString(R.string.systemmessage) + response.body().getMessage(), Toast.LENGTH_SHORT).show();
//                                }
//                            }
//
//                        }
//                    }
//                    else Toast.makeText(SplashActivity.this,getString(R.string.systemmessage)+response.errorBody(),Toast.LENGTH_SHORT).show();
//
//                }
//                catch (Exception e){
//                    Log.e(TAG, "onResponse:110 "+e );
//                }
//            }
//
//            @Override
//            public void onFailure(Call<AppOpenModel> call, Throwable t) {
//                Toast.makeText(SplashActivity.this,getString(R.string.systemmessage)+t,Toast.LENGTH_SHORT).show();
//            }
//        });
//    }

    @Override
    protected void onPause() {
        super.onPause();
        dismissProgressDialog();

    }

    @Override
    protected void onStop() {
        super.onStop();
        dismissProgressDialog();

    }

    private void dismissProgressDialog() {
        if (progressDialog != null && progressDialog.isShowing()) {
            progressDialog.dismiss();
        }
    }

    public static boolean isNetworkAvailable(Context c) {
        ConnectivityManager connectivityManager = (ConnectivityManager) c
                .getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo activeNetworkInfo = connectivityManager
                .getActiveNetworkInfo();
        return activeNetworkInfo != null && activeNetworkInfo.isConnected();
    }



    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
//        if(requestCode==105){
//            Intent intent = new Intent(SplashActivity.this, MainActivity.class);
//            startActivity(intent);
//            finish();
//        }
    }




}

File: BidWinKoModify/app/src/main/java/com/bidwinko/screens/activity/SplashActivity.kt
package com.bidwinko.screens.activity

import android.app.Activity
import android.app.ActivityOptions
import android.app.AlertDialog
import android.app.ProgressDialog
import android.content.Context
import android.content.Intent
import android.content.pm.PackageInfo
import android.content.pm.PackageManager
import android.net.ConnectivityManager
import android.net.Uri
import android.os.Build
import android.os.Bundle
import android.os.Handler
import android.provider.Settings
import android.util.Log
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityOptionsCompat
import com.bidwinko.MainActivity
import com.bidwinko.R
import com.bidwinko.model.RequestModels.AppOpenRequest
import com.bidwinko.utilies.Constants
import com.bidwinko.utilies.SessionManager
import com.bidwinko.viewModel.mainViewModel
import com.google.android.gms.ads.identifier.AdvertisingIdClient
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.Job
import kotlinx.coroutines.async
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext

private val coroutineScope = CoroutineScope(Dispatchers.Main + Job())

private var handler: Handler? = null
private lateinit var adInfoData: String
private lateinit var mainViewModel: mainViewModel
private lateinit var sessionManager: SessionManager
lateinit var versionName: String

class SplashActivity() : AppCompatActivity() {
    var progressDialog: ProgressDialog? = null
    var TAG = "Splash Screen"
    var tranTxt = ""
    var txtVersion: TextView? = null
    override fun onStart() {
        super.onStart()
        Log.e("SplashActivity", "openig SplashActivity")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.e("SplashActivity", "in OnDestroy  SplashActivity")
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        txtVersion = findViewById(R.id.txtvesion)
        supportActionBar?.hide()
        mainViewModel = mainViewModel(this)
        sessionManager = SessionManager(this)
        val logo_image = findViewById<ImageView>(R.id.image)
        val logo_name = findViewById<TextView>(R.id.name)

//        image.startAnimation(AnimationUtils.loadAnimation(this,R.anim.zoom_out))
        coroutineScope.launch {
            try {
                val adInfo = async { getAdvertisingIdInfo() }
                val packageInfo = async { getPackageInfo() }

                adInfoData = adInfo.await()?.id ?: ""
                val versionCode = packageInfo.await().versionCode
                versionName = packageInfo.await().versionName
                txtVersion?.let { it.setText("Version: $versionName") }
                val deviceId =
                    Settings.Secure.getString(contentResolver, Settings.Secure.ANDROID_ID)
                sessionManager.InitializeValue(Constants.DEVICE_ID, deviceId)
                val deviceType = Build.MODEL
                val deviceManufacturer = Build.MANUFACTURER
                val deviceName = "$deviceManufacturer $deviceType"
                Log.e(
                    "SplashScreen",
                    "$deviceId, $deviceType, $deviceName, $adInfoData, $versionName, $versionCode"
                )
                sessionManager.InitializeValue(Constants.VERSION_CODE, versionCode.toString())
                sessionManager.InitializeValue(Constants.VERSION_NAME, versionName)
                sessionManager.InitializeValue(Constants.DEVICE_NAME, deviceName)
                sessionManager.InitializeValue(Constants.DEVICE_TYPE, deviceType)
                sessionManager.InitializeValue(Constants.ADVERTISING_ID, adInfoData)

                if (isNetworkAvailable(this@SplashActivity)) {
                    if (!sessionManager.GetValue(Constants.USER_ID).equals("null") &&
                        !sessionManager.GetValue(Constants.SECURITY_TOKEN).equals("null")
                    ) {
                        Log.e("SplashSCreen", "openig OpenApp API")
                        openApp(this@SplashActivity)
                    } else {
                        Handler().postDelayed({
                            if (!(this@SplashActivity as Activity).isFinishing) {
                                progressDialog = ProgressDialog(this@SplashActivity)
                                progressDialog!!.setMessage(getString(R.string.loadingwait))
                                progressDialog!!.show()
                                progressDialog!!.setCancelable(false)
                            }
                            Log.e("SplashSCreen", "sending to google signin activity !")
                            val intent = Intent(this@SplashActivity, GoogleLoginActivity::class.java)
                            val options = ActivityOptions.makeSceneTransitionAnimation(this@SplashActivity,
                                android.util.Pair(logo_image, "logo"),
                                android.util.Pair(logo_name,"appname"))
                            startActivity(intent,options.toBundle())
                            overridePendingTransition(R.anim.fast_right_slide_in, R.anim.fast_left_slide_out)
                            finish()
                        }, 3000)
                    }
                } else {
                    val builder = AlertDialog.Builder(this@SplashActivity)
                    builder.setTitle("Alert!")
                    builder.setMessage("Please Check Your Internet Connection.")
                    builder.setPositiveButton(
                        "Exit"
                    ) { dialog, which ->
                        dialog.dismiss()
                        finish()
                    }
                    builder.setCancelable(false)
                    builder.show()
                }

            } catch (e: Exception) {
                Toast.makeText(this@SplashActivity, "Restart Your App !!", Toast.LENGTH_LONG).show()
            }
        }
    }


    private fun openApp(context : Context) {
        if (this@SplashActivity.isFinishing) {
            val progressDialog = ProgressDialog(this@SplashActivity)
            progressDialog.setMessage(getString(R.string.loadingwait))
            progressDialog.show()
            progressDialog.setCancelable(false)
        }

        val versionname = sessionManager.GetValue(Constants.VERSION_NAME)
        val versionCode = sessionManager.GetValue(Constants.VERSION_CODE)
        val UserId = sessionManager.GetValue(Constants.USER_ID)
        val securityToken = sessionManager.GetValue(Constants.SECURITY_TOKEN)

        val appOpenRequest = AppOpenRequest(
            userId = UserId,
            securityToken = securityToken,
            versionName = versionname,
            versionCode = versionCode
        )

        mainViewModel.appOpen(appOpenRequest).observe(this@SplashActivity) {
            dismissProgressDialog()
            if (it.status == 200) {

                SessionManager(this).InitializeValue(Constants.TOTAL_BIDS,it.bids)
                if (it.forceUpdate) {
                    startActivity(
                        Intent(
                            Intent.ACTION_VIEW,
                            Uri.parse("https://play.google.com/store/apps/details?id=$packageName")
                        )
                    )
                    finish()
                } else {
                    SessionManager(this).InitializeValue(Constants.APP_URL,it.appUrl)
                    val intent = Intent(this, MainActivity::class.java)
                    startActivity(intent)
                    overridePendingTransition(R.anim.fast_right_slide_in, R.anim.fast_left_slide_out)
                    finish()
                }
            } else {
                Toast.makeText(this@SplashActivity, "Somthing Went Wrong Try Again Later!!", Toast.LENGTH_SHORT).show()
            }
        }
    }

    override fun onPause() {
        super.onPause()
        dismissProgressDialog()
    }

    override fun onStop() {
        super.onStop()
        dismissProgressDialog()
    }

    private fun dismissProgressDialog() {
        if (progressDialog != null && progressDialog!!.isShowing) {
            progressDialog!!.dismiss()
        }
    }
    companion object {
        fun isNetworkAvailable(c: Context): Boolean {
            val connectivityManager = c
                .getSystemService(CONNECTIVITY_SERVICE) as ConnectivityManager
            val activeNetworkInfo = connectivityManager
                .activeNetworkInfo
            return activeNetworkInfo != null && activeNetworkInfo.isConnected
        }
    }

    private suspend fun getPackageInfo(): PackageInfo {
        return withContext(Dispatchers.IO) {
            try {
                packageManager.getPackageInfo(packageName, 0)
            } catch (e: PackageManager.NameNotFoundException) {
                e.printStackTrace()
                PackageInfo()
            }
        }
    }

    private suspend fun getAdvertisingIdInfo(): AdvertisingIdClient.Info? {
        return withContext(Dispatchers.IO) {
            try {
                AdvertisingIdClient.getAdvertisingIdInfo(applicationContext)
            } catch (e: Exception) {
                e.printStackTrace()
                null
            }
        }
    }



}
